src/Core/Domain/Product/ValueObject/ProductRedirectOption.php

```diff
@@ -0,0 +1,227 @@
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\ValueObject;
+
+use LogicException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use PrestaShop\PrestaShop\Core\Product\ProductInterface;
+
+/**
+ * Holds values for product redirect option.
+ * todo: there are type constants declared already in @var ProductInterface. Should I reuse them? deprecate them?
+ */
+class ProductRedirectOption
+{
+    /**
+     * Represents value of redirect target id when NO_REDIRECT type is provided
+     */
+    const NO_TARGET_VALUE = 0;
+
+    /**
+     * Represents value of no redirection. Page not found (404) will be displayed.
+     */
+    const TYPE_NO_REDIRECT = '404';
+
+    /**
+     * Represents value of permanent redirection to a category
+     */
+    const TYPE_CATEGORY_PERMANENT = '301-category';
+
+    /**
+     * Represents value of temporary redirection to a category
+     */
+    const TYPE_CATEGORY_TEMPORARY = '302-category';
+
+    /**
+     * Represents value of permanent redirection to another product
+     */
+    const TYPE_PRODUCT_PERMANENT = '301-product';
+
+    /**
+     * Represents value of temporary redirection to another product
+     */
+    const TYPE_PRODUCT_TEMPORARY = '302-product';
+
+    /**
+     * Available redirection types
+     */
+    const AVAILABLE_REDIRECT_TYPES = [
+        self::TYPE_NO_REDIRECT => self::TYPE_NO_REDIRECT,
+        self::TYPE_CATEGORY_PERMANENT => self::TYPE_CATEGORY_PERMANENT,
+        self::TYPE_CATEGORY_TEMPORARY => self::TYPE_CATEGORY_TEMPORARY,
+        self::TYPE_PRODUCT_PERMANENT => self::TYPE_PRODUCT_PERMANENT,
+        self::TYPE_PRODUCT_TEMPORARY => self::TYPE_PRODUCT_TEMPORARY,
+    ];
+
+    /**
+     * @var string
+     */
+    private $redirectType;
+
+    /**
+     * @var int
+     */
+    private $redirectTargetId;
+
+    /**
+     * Builds self with a type of no_redirect, automatically filling the only available target id value for this type
+     *
+     * @return static
+     */
+    public static function buildNoRedirect(): self
```

Comment:
I see you're starting to put these static constructors everywhere 😄 
Nice ! it's indeed a very good practice